executor/aggregate.go

```diff
@@ -226,8 +231,85 @@ func (e *StreamAggExec) Next(goCtx goctx.Context) (Row, error) {
 	return retRow, nil
 }
 
+// NextChunk implements the Executor NextChunk interface.
+func (e *StreamAggExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+
+	if e.executed {
+		return nil
+	}
+
+	mutableRow := chunk.MutRowFromTypes(e.Schema().GetTypes())
+	retRow := make([]types.Datum, 0, e.Schema().Len())
+
+	for {
+		if e.childCursor < e.childrenResults[0].NumRows() {
+			for row := e.childrenResults[0].GetRow(e.childCursor); row != e.childrenResults[0].End(); row = row.Next() {
+				meetNewGroup, err := e.meetNewGroup(row)
+				if err != nil {
+					e.executed = true
+					return errors.Trace(err)
+				}
+				if meetNewGroup {
+					retRow = retRow[:0]
+					for i, af := range e.AggFuncs {
+						retRow = append(retRow, af.GetResult(e.aggCtxs[i]))
+						e.aggCtxs[i] = af.CreateContext()
+					}
+					mutableRow.SetDatums(retRow...)
+					if chk.NumCols() == 0 {
+						chk.SetNumVirtualRows(chk.NumRows() + 1)
+					} else {
+						chk.AppendRow(0, mutableRow.ToRow())
+					}
+				}
+				for i, af := range e.AggFuncs {
+					err := af.Update(e.aggCtxs[i], e.StmtCtx, row)
+					if err != nil {
+						e.executed = true
+						return errors.Trace(err)
+					}
+				}
+				if meetNewGroup && chk.NumRows() >= e.maxChunkSize {
+					e.childCursor = row.Idx() + 1
+					return nil
+				}
+			}
+		}
+
+		// Get resource rows from child.
+		err := e.children[0].NextChunk(goCtx, e.childrenResults[0])
+		if err != nil {
+			e.executed = true
+			return errors.Trace(err)
+		}
+		e.childCursor = 0
+
+		// No more data.
+		if e.childrenResults[0].NumRows() == 0 {
+			e.executed = true
+			if !e.hasData && len(e.GroupByItems) != 0 {
+				return nil
+			}
+			retRow = retRow[:0]
+			for i, af := range e.AggFuncs {
+				retRow = append(retRow, af.GetResult(e.aggCtxs[i]))
+				e.aggCtxs[i] = af.CreateContext()
+			}
+			mutableRow.SetDatums(retRow...)
+			if chk.NumCols() == 0 {
+				chk.SetNumVirtualRows(chk.NumRows() + 1)
+			} else {
+				chk.AppendRow(0, mutableRow.ToRow())
+			}
```

Comment:
Could we extract line294-304 into a function? It seems duplicate with line 254-264.